proto/v1beta1/spdx.proto

```diff
@@ -277,115 +405,10 @@ message RelationshipOccurrence {
-    ANCESTOR_OF = 17;
-    // Is to be used when SPDXRef-A is a descendant of (same lineage but
-    // postdates) SPDXRef-B
-    DESCENDANT_OF = 18;
-    // Is to be used when SPDXRef-A is a variant of (same lineage but not
-    // clear which came first) SPDXRef-B
-    VARIANT_OF = 19;
-    // Is to be used when distributing SPDXRef-A requires that SPDXRef-B also
-    // be distributed
-    DISTRIBUTION_ARTIFACT = 20;
-    // Is to be used when SPDXRef-A is a patch file for (to be applied to)
-    // SPDXRef-B
-    PATCH_FOR = 21;
-    // Is to be used when SPDXRef-A is a patch file that has been applied to
-    // SPDXRef-B
-    PATCH_APPLIED = 22;
-    // Is to be used when SPDXRef-A is an exact copy of SPDXRef-B
-    COPY_OF = 23;
-    // Is to be used when SPDXRef-A is a file that was added to SPDXRef-B
-    FILE_ADDED = 24;
-    // Is to be used when SPDXRef-A is a file that was deleted from SPDXRef-B
-    FILE_DELETED = 25;
-    // Is to be used when SPDXRef-A is a file that was modified from SPDXRef-B
-    FILE_MODIFIED = 26;
-    // Is to be used when SPDXRef-A is expanded from the archive SPDXRef-B
-    EXPANDED_FROM_ARCHIVE = 27;
-    // Is to be used when SPDXRef-A dynamically links to SPDXRef-B
-    DYNAMIC_LINK = 28;
-    // Is to be used when SPDXRef-A statically links to SPDXRef-B
-    STATIC_LINK = 29;
-    // Is to be used when SPDXRef-A is a data file used in SPDXRef-B
-    DATA_FILE_OF = 30;
-    // Is to be used when SPDXRef-A is a test case used in testing SPDXRef-B
-    TEST_CASE_OF = 31;
-    // Is to be used when SPDXRef-A is used to build SPDXRef-B
-    BUILD_TOOL_OF = 32;
-    // Is to be used when SPDXRef-A is used as a development tool for
-    // SPDXRef-B
-    DEV_TOOL_OF = 33;
-    // Is to be used when SPDXRef-A is used for testing SPDXRef-B
-    TEST_OF = 34;
-    // Is to be used when SPDXRef-A is used as a test tool for SPDXRef-B
-    TEST_TOOL_OF = 35;
-    // Is to be used when SPDXRef-A provides documentation of SPDXRef-B
-    DOCUMENTATION_OF = 36;
-    // Is to be used when SPDXRef-A is an optional component of SPDXRef-B
-    OPTIONAL_COMPONENT_OF = 37;
-    // Is to be used when SPDXRef-A is a metafile of SPDXRef-B
-    METAFILE_OF = 38;
-    // Is to be used when SPDXRef-A is used as a package as part of SPDXRef-B
-    PACKAGE_OF = 39;
-    // Is to be used when (current) SPDXRef-DOCUMENT amends the SPDX
-    // information in SPDXRef-B
-    AMENDS = 40;
-    // Is to be used when SPDXRef-A is a prerequisite for SPDXRef-B
-    PREREQUISITE_FOR = 41;
-    // Is to be used when SPDXRef-A has as a prerequisite SPDXRef-B
-    HAS_PREREQUISITE = 42;
-    // Is to be used for a relationship which has not been defined in the
-    // formal SPDX specification. A description of the relationship should be
-    // included in the Relationship comments field
-    OTHER = 43;
-  }
 
   // The type of relationship between the source and target SPDX elements
-  Type type = 3;
+  RelationshipType type = 3;
```

Comment:
wouldn't the reference to the Note it's attached to be sufficient in this case? (perhaps here as well, an example might help the review)

Reply:
The reason for having the type in the occurrence is that it makes it possible to get all information by calling ListOccurrences without needing to call ListNotes. It is an optimization and for convenience. We do something similar in vulnerability occurrences.

Here is an example:
```
// RelationshipNote
{
    name: "projects/test-project/notes/345",
    short_description: "Package dependency",
    kind: "SPDX_RELATIONSHIP",
    spdx_relationship: {
        type: "DEPENDS_ON"
    }
}

// RelationshipOccurrence
{
    name: "projects/test-project/occurrences/678",
    resource: "https://us-west1.docker.pkg.dev/test-project/my-repo/debian/sha256:12345",
    note_name: "projects/test-project/notes/345",
    kind: "SPDX_RELATIONSHIP",
    spdx_package: {
        source: "projects/test-project/occurrences/345",
        target: "projects/test-project/occurrences/349", // package dependency
        type: "DEPENDS_ON"
    }
}
```